tests/e2e/framework/multicluster.go

```diff
@@ -52,62 +52,51 @@ func getKubeConfigFromFile(dirname string) (string, error) {
 	return remoteKube, nil
 }
 
-func (k *KubeInfo) generateRemoteIstio(src, dst string) error {
-	content, err := ioutil.ReadFile(src)
-	if err != nil {
-		log.Errorf("cannot read remote yaml file %s", src)
-		return err
-	}
+func (k *KubeInfo) getEndpointIPForService(svc string) (ip string, err error) {
 	getOpt := meta_v1.GetOptions{IncludeUninitialized: true}
-	var statsdEPS, pilotEPS, mixerEPS *v1.Endpoints
-
+	var eps *v1.Endpoints
+	// Wait until endpoint is obtained
 	for i := 0; i <= 200; i++ {
-		pilotEPS, err = k.KubeClient.CoreV1().Endpoints(k.Namespace).Get("istio-pilot", getOpt)
-		if (len(pilotEPS.Subsets) != 0) && (err == nil) {
-			mixerEPS, err = k.KubeClient.CoreV1().Endpoints(k.Namespace).Get("istio-policy", getOpt)
-			if (len(mixerEPS.Subsets) != 0) && (err == nil) {
-				statsdEPS, err = k.KubeClient.CoreV1().Endpoints(k.Namespace).Get("istio-statsd-prom-bridge", getOpt)
-				if (len(statsdEPS.Subsets) != 0) && (err == nil) {
-					break
-				}
-			}
+		eps, err = k.KubeClient.CoreV1().Endpoints(k.Namespace).Get(svc, getOpt)
+		if (len(eps.Subsets) == 0) || (err != nil) {
+			time.Sleep(time.Second * 1)
+		} else {
+			break
 		}
-		time.Sleep(time.Second * 1)
-	}
-	if err != nil {
-		err = fmt.Errorf("could not get endpoints from local cluster")
-		return err
 	}
```

Comment:
I think you have the same flaw that the original code had where you don't check if you exited the loop after 200 seconds which should be an error condition.

Reply:
oh see line 68 - 76, which should have that case covered since no valid ip will be found.